### PR TITLE
Update doc according npm workspace

### LINKED ISSUE #609: The CLI isn't working
On the `develop` branch (commit `c0f27cf`), running `npm start -- --task cifar10 --numberOfUsers 2 --epochs 2 --roundDuration 5` triggers consecutive `module not found` errors.
It seems many dependencies are missing from the CLI `package.json`, such as tfjs, tfjs-node, axios, etc.

### LINKED ISSUE #635: Update module installation doc
The installation instructions are currently outdated, especially now that we are using npm workspaces. 
Update how to install and how to use each module of disco.

- [x] misformatted markdown at the top of DEV.md
- [x] outdated testing instructions in docs/ONBOARDING.md
- [x] outdated VSCode import tip in docs/ONBOARDING.md
- [x] docs/node_example
- [x] web-client/README.md
- [ ] server/README.md
- [x] cli/README.md
- [ ] discojs/README.md
- [ ] discojs/discojs-core/README.md
- [ ] discojs/discojs-node/README.md
- [ ] discojs/discojs-web/README.md
- [ ] Merge server, web-client, discojs installation and usage in DEV.md

### PR DESCRIPTION
**Pull request by JulienVig**
* Fixes #635 
  Update each module's instruction now that npm workspaces has been integrated

I reorganized the documentation with 3 different types of DISCO users in mind. What they should be able to do are concentric subsets:
1. non-technical users → want to use the UI to train and create their own custom tasks without coding
2. technical users → additionally code in Node.js to train and create their own custom tasks with discojs-node and discojs-server
3. contributors → additionally need to be able to build, test, modify and learn in-depth information

As such, I organized the doc by following these successive levels:
1. README.md for everyone and especially non-technical users
2. DEV.md for technical users and contributors, it now contains instructions for the different ways to use DISCO
3. CONTRIBUTING.md for contributors, with build, test, coding conventions and in-depth information needed to modify the code base.

Additionally:
- Removed redundant module readme (discojs, disco-core/node/web, web-client) as instructions are now centralized in DEV.md and CONTRIBUTING.md
- Merged ONBOARDING.md into CONTRIBUTING.md and deleted the former. I didn't find the difference between the two files self-explanatory enough and contributors have to go through both of them anyway so I merged the two into one file.
- Created a new example for adding a custom task to the server in `docs/node_example`, which I renamed `docs/examples` with the objective of adding further examples in the future.

Out of scope: #639 Adding custom tasks is currently very convoluted and needs a rework to make it easier for external users. Information on how to add custom tasks is not clear and not accessible enough for any of the three user categories I mentioned.

* Also fixes #609 
  I expect it got fixed by the previous PRs.

### CODE REVIEW

File: DEV.md [hunk 1]
```diff
@@ -125,15 +124,58 @@ npm -w web-client start # from another terminal
...
-> Make sure to first start the server to ensure that it is listening to port 8080.
-
 **You can now access DISCO at http://localhost:8081/**
 
+## How to use DISCO
+
+There are multiple ways to use and interact with DISCO, depending on your objective:
+
+- A non-technical user that wants to train models in a distributed manner without coding would want to use DISCO through the `web-client`. To do so, starting a local `server` instance is also needed as a backend to the `web-client`. Similarly, a contributor aiming to implement new UI features would certainly want to run the same setup.
+- A technical user may find it more flexible to use DISCO from a Node.js script, which gives users a finer control over the process. The `discojs-node` module is tailored to be used in Node.js scripts and allows to load data, start a server and run distributed machine learning training tasks.
```
Comment (tharvik) [inline DEV.md]: starting a server is in the `server` package only, but indeed `discojs-node` can help _build_ a server

File: cli/package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "@epfml/disco-cli",
   "private": true,
-  "main": "dist/benchmark.ts",
+  "main": "dist/cli.ts",
```
Comment (tharvik) [inline cli/package.json]: haa oupsi, this never worked, as "dist" contains only javascript files

```suggestion
  "main": "dist/cli.js",
```

File: cli/src/args.ts [hunk 3]
```diff
@@ -19,21 +18,21 @@ type BenchmarkUnsafeArguments = {
...
     roundDuration: { type: Number, alias: 'r', description: 'Round duration', defaultValue: 10 },
-    batchSize: { type: Number, alias: 'b', description: 'Round duration', defaultValue: 10 },
+    batchSize: { type: Number, alias: 'b', description: 'Training batch size', defaultValue: 10 },
     save: { type: Boolean, alias: 's', description: 'Save logs of benchmark', defaultValue: false },
     help: { type: Boolean, optional: true, alias: 'h', description: 'Prints this usage guide' }
   },
   {
     helpArg: 'help',
-    headerContentSections: [{ header: 'Disco benchmark', content: 'npm run benchmark -- [Options]\n' + argExample }]
+    headerContentSections: [{ header: 'DISCO CLI', content: 'npm run start -- [Options]\n' + argExample }]
```
Comment (tharvik) [inline cli/src/args.ts]: both are valid but consistency is good

```suggestion
    headerContentSections: [{ header: 'DISCO CLI', content: 'npm start -- [Options]\n' + argExample }]
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,189 +1,232 @@
...
 
-We use the testing framework [mocha](https://mochajs.org/) and the [chai](https://www.chaijs.com/) assertion library.
-To write tests for a code file `<filename>.ts`, create a `<filename>.spec.ts` file. This way, tests will be executed automatically for continuous integration.
-Each part of DISCO has a respective test suite:
-* `discojs` needs to have a `server` instance to be tested:
 ```
-cd server
-npm start
+npm -w web-client start # from the root folder
+npm start # from the web-client folder
```
Comment (tharvik) [inline docs/CONTRIBUTING.md:82]: hum, sometimes commands are from the root folder, sometimes from the local dir. we need to keep to only one style of command.

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -1,189 +1,232 @@
...
+```
+
+The Vue development mode supports hot-reloading via `nodemon` and the client will automatically restart whenever a change in `web-client` is detected. Starting the Vue client should print something similar to
```
Comment (tharvik) [inline docs/CONTRIBUTING.md]: haa, in a perfect world, it should, but `vue-cli-service` already do a pretty good job at watching changes and hot-reloading modules so I kept it.

```suggestion
The Vue development mode supports hot-reloading via `vue-cli-service` and the client will automatically restart whenever a change in `web-client` is detected. Starting the Vue client should print something similar to
```

File: docs/examples/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+- How to train a federated model on pre-defined tasks
+- How to create multiple clients
+- How to start a server instance from a script
+- How to load and preprocess local data
+
+You can run the training example as follows:
+
+```
+cd docs/example
```
Comment (tharvik) [inline docs/examples/README.md:17]:
```suggestion
cd docs/examples
```

File: docs/examples/README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+### Adding a new custom task
+
+The server handles the pre-defined machine learning tasks and sends them to users when queried. `custom_task.ts` illustrates how to add a custom task to the server such that it can be queried by users. More details on custom tasks can be found in [TASK.md](../TASK.md).
+
+You can run the custom task example with:
+
+```
+cd docs/example
```
Comment (tharvik) [inline docs/examples/README.md]:
```suggestion
cd docs/examples
```

File: docs/examples/tsconfig.json
Comment (tharvik) [inline docs/examples/tsconfig.json:1]: add `compilerOptions.outDir = "dist"` to generate output in the "dist" subfolder, npm' scripts are looking there

File: docs/examples/package.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    "test": ": nothing"
+  },
+  "license": "ISC",
+  "dependencies": {
+    "@epfml/disco-server": "*",
+    "@epfml/discojs-node": "*"
+  },
+  "devDependencies": {
+    "ts-node": "10"
+  }
```
Comment (tharvik) [inline docs/examples/package.json]: not needed to run anything here

```suggestion
```

File: docs/examples/README.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+npm run custom_Task # compiles TypeScript and runs custom_task.ts
```
Comment (tharvik) [inline docs/examples/README.md]:
```suggestion
npm run custom_task # compiles TypeScript and runs custom_task.ts
```

File: docs/examples/tsconfig.eslint.json
Comment (tharvik) [inline docs/examples/tsconfig.eslint.json:1]: do we add linting support here at all? if feel a bit out of scope. the current `npm run lint` doesn't work (but can made to work if merge the root eslintrc with the local one)

**Review by tharvik — APPROVED** [review]
superbe, thanks for the tiding up :1st_place_medal:

comments are not blocking, rather "nice to have"

File: docs/CONTRIBUTING.md (hunk 4 above)
Comment (JulienVig) [reply docs/CONTRIBUTING.md:82]: In this case I'm giving two alternative ways of starting the web-client depending on the cwd. I'm trying to always use -w but I'm sometimes wary that some readers may have missed how -w worked

File: docs/CONTRIBUTING.md (hunk 5 above)
Comment (JulienVig) [reply docs/CONTRIBUTING.md]: Ah nice thanks for reading attentively!

File: docs/examples/tsconfig.eslint.json
Comment (JulienVig) [reply docs/examples/tsconfig.eslint.json:1]: I think that would be nice, how can I do it? I'm not what you mean by "merge tje root eslintrc"

Comment (tharvik) [reply docs/examples/tsconfig.eslint.json:1]: eslint uses a cascading config organisation, meaning that it aggregates all eslintrc files start from cwd and bubbling up the parent chain.
in the current case, it takes docs/examples/.eslintrc.json & .eslintrc.js, making the example non-local. you can copy the one from root in docs/examples and change its env to use node instead of mocha.

File: docs/CONTRIBUTING.md (hunk 4 above)
Comment (tharvik) [reply docs/CONTRIBUTING.md:82]: your call :)
